app/models/page.rb

```diff
@@ -27,7 +27,7 @@ class Page < ActiveRecord::Base
   validates :liquid_layout, presence: true
   validates :publish_status, presence: true
   validate  :primary_image_is_owned
-  validates :canonical_url, allow_blank: true, format: { with: %r{\Ahttps{0,1}:\/\/.+\..+} }
+  validates :canonical_url, allow_blank: true, format: { with: %r{\Ahttps{0,1}:\/\/.+\..+\Z} }
```

Comment:
You probably want `\z` instead of `\Z` as the uppercase variant allows newline.

```ruby
"https://something.foo\n".match(%r{\Ahttps{0,1}:\/\/.+\..+\Z})
#<MatchData "https://something.foo">

"https://something.foo\n".match(%r{\Ahttps{0,1}:\/\/.+\..+\z})
=> nil
```